### PR TITLE
Fix: Missing Connection Successful Logs

### LINKED ISSUE #1204: Missing Successful Connection Logs At Info Level
**Describe the bug**
The logs are improper for successful connection at INFO Level.

**Expected behavior**
The logs must be consistent across all databases.

### PR DESCRIPTION
**Pull request by coolwednesday**
## Pull Request Template

**Description:**

-   Closes #1204 
-  Corrected log level of the connection logs in datasources.
-  We have also removed [build -exclude](go:build exclude) as we need the test to be checked in the pipeline. 

**Checklist:**

-   [x] I have formatted my code using  `goimport`  and  `golangci-lint`.
-   [x] All new code is covered by unit tests.
-   [x] This PR does not decrease the overall code coverage.
-   [x] I have reviewed the code comments and documentation for clarity.

**Thank you for your contribution!**

### CODE REVIEW

**Comment by ccoVeille** [comment]
Why closing?

**Comment by ccoVeille** [comment]
Is it because this one superseded it?

https://github.com/gofr-dev/gofr/pull/1205

**Comment by coolwednesday** [comment]
> Why closing?

There are continuous build fails. I will address that and then reopen, this PR.

File: pkg/gofr/datasource/file/s3/fs.go [hunk 1]
```diff
@@ -109,7 +111,7 @@ func (f *fileSystem) Connect() {
 	st = statusSuccess
 	msg = "S3 Client connected."
 
-	f.logger.Logf("Connected to S3 bucket %s", f.config.BucketName)
+	f.logger.Infof("Connected to S3 bucket %s", f.config.BucketName)
```
Comment (Umang01-hash) [inline pkg/gofr/datasource/file/s3/fs.go:114]: here also we should start with a small case character

File: pkg/gofr/datasource/file/ftp/fs.go [hunk 2]
```diff
@@ -84,6 +84,8 @@ func (f *fileSystem) Connect() {
 
 	ftpServer := fmt.Sprintf("%v:%v", f.config.Host, f.config.Port)
 
+	f.logger.Infof("Connecting to FTP server at '%v'", ftpServer)
```
Comment (Umang01-hash) [inline pkg/gofr/datasource/file/ftp/fs.go:87]: We should start with a small-case character for logging here just like we did for other datasources

File: pkg/gofr/datasource/file/ftp/fs.go [hunk 3]
```diff
@@ -116,7 +118,7 @@ func (f *fileSystem) Connect() {
 
 	status = "LOGIN SUCCESS"
 
-	f.logger.Logf("Connected to FTP server at '%v'", ftpServer)
+	f.logger.Infof("Connected to FTP server at '%v'", ftpServer)
```
Comment (Umang01-hash) [inline pkg/gofr/datasource/file/ftp/fs.go:121]: Log should be started with a small case character here also

File: pkg/gofr/datasource/kv-store/badger/mock_logger.go [hunk 4]
```diff
@@ -1,9 +1,12 @@
-// Code generated by MockGen. DO NOT EDIT.
+//go:build exclude
+
+//
+//Code generated by MockGen. DO NOT EDIT.
```
Comment (ccoVeille) [inline pkg/gofr/datasource/kv-store/badger/mock_logger.go:4]:
```suggestion
// Code generated by MockGen. DO NOT EDIT.
```

File: pkg/gofr/datasource/pubsub/kafka/mock_interfaces.go [hunk 5]
```diff
@@ -1,4 +1,4 @@
-// Code generated by MockGen. DO NOT EDIT.
+//Code generated by MockGen. DO NOT EDIT.
```
Comment (ccoVeille) [inline pkg/gofr/datasource/pubsub/kafka/mock_interfaces.go:1]: I don't get how this change was introduced

Do you have a new version of mockgen?

File: pkg/gofr/datasource/pubsub/nats/client.go [hunk 6]
```diff
@@ -37,6 +37,8 @@ func (c *Client) Connect() error {
 		return err
 	}
 
+	c.logger.Log("connecting to NATS sever at %v", c.Config.Server)
```
Comment (ccoVeille) [inline pkg/gofr/datasource/pubsub/nats/client.go:40]:
```suggestion
	c.logger.Log("connecting to NATS server at %v", c.Config.Server)
```

😁

**Review by Umang01-hash — APPROVED** [review]

File: pkg/gofr/datasource/cassandra/cassandra.go [hunk 7]
```diff
@@ -62,7 +62,7 @@ func New(conf Config) *Client {
 
 // Connect establishes a connection to Cassandra and registers metrics using the provided configuration when the client was Created.
 func (c *Client) Connect() {
-	c.logger.Logf("connecting to cassandra at %v on port %v to keyspace %v", c.config.Hosts, c.config.Port, c.config.Keyspace)
+	c.logger.Debugf("connecting to cassandra at %v on port %v to keyspace %v", c.config.Hosts, c.config.Port, c.config.Keyspace)
```
Comment (ccoVeille) [inline pkg/gofr/datasource/cassandra/cassandra.go:65]:
```suggestion
	c.logger.Debugf("connecting to Cassandra at %v on port %v to keyspace %v", c.config.Hosts, c.config.Port, c.config.Keyspace)
```

File: pkg/gofr/datasource/cassandra/cassandra_test.go [hunk 8]
```diff
@@ -82,7 +82,7 @@ func Test_Connect(t *testing.T) {
 
 	cassandraBuckets := []float64{.05, .075, .1, .125, .15, .2, .3, .5, .75, 1, 2, 3, 4, 5, 7.5, 10}
 
-	mockLogger.EXPECT().Logf("connecting to cassandra at %v on port %v to keyspace %v", "host1", 9042, "test_keyspace")
+	mockLogger.EXPECT().Debugf("connecting to cassandra at %v on port %v to keyspace %v", "host1", 9042, "test_keyspace")
```
Comment (ccoVeille) [inline pkg/gofr/datasource/cassandra/cassandra_test.go:85]:
```suggestion
	mockLogger.EXPECT().Debugf("connecting to Cassandra at %v on port %v to keyspace %v", "host1", 9042, "test_keyspace")
```

Maybe everywhere in fact

File: pkg/gofr/datasource/clickhouse/clickhouse.go [hunk 9]
```diff
@@ -70,7 +70,7 @@ func (c *client) UseTracer(tracer any) {
 func (c *client) Connect() {
 	var err error
 
-	c.logger.Logf("connecting to clickhouse db at %v to database %v", c.config.Hosts, c.config.Database)
+	c.logger.Debugf("connecting to clickhouse db at %v to database %v", c.config.Hosts, c.config.Database)
```
Comment (ccoVeille) [inline pkg/gofr/datasource/clickhouse/clickhouse.go:73]:
```suggestion
	c.logger.Debugf("connecting to ClickHouse db at %v to database %v", c.config.Hosts, c.config.Database)
```

File: pkg/gofr/datasource/file/ftp/mock_interface.go [hunk 10]
```diff
@@ -1,6 +1,6 @@
 //go:build exclude
 
-// Code generated by MockGen. DO NOT EDIT.
+//Code generated by MockGen. DO NOT EDIT.
```
Comment (ccoVeille) [inline pkg/gofr/datasource/file/ftp/mock_interface.go:3]: Nope

File: pkg/gofr/datasource/mongo/mongo.go [hunk 11]
```diff
@@ -78,7 +78,7 @@ func (c *Client) UseTracer(tracer any) {
 
 // Connect establishes a connection to MongoDB and registers metrics using the provided configuration when the client was Created.
 func (c *Client) Connect() {
-	c.logger.Logf("connecting to mongoDB at %v to database %v", c.config.URI, c.config.Database)
+	c.logger.Debug(fmt.Sprintf("connecting to mongoDB at %v to database %v", c.config.URI, c.config.Database))
```
Comment (ccoVeille) [inline pkg/gofr/datasource/mongo/mongo.go:81]:
```suggestion
	c.logger.Debugf("connecting to mongoDB at %v to database %v", c.config.URI, c.config.Database)
```

Comment (ccoVeille) [inline pkg/gofr/datasource/mongo/mongo.go:81]:
```suggestion
	c.logger.Debug(fmt.Sprintf("connecting to MongoDB at %v to database %v", c.config.URI, c.config.Database))
```

File: pkg/gofr/datasource/pubsub/nats/client.go [hunk 12]
```diff
@@ -33,6 +33,8 @@ type messageHandler func(context.Context, jetstream.Msg) error
 
 // Connect establishes a connection to NATS and sets up jStream.
 func (c *Client) Connect() error {
+	c.logger.Debugf("connecting to NATS server at %v", c.Config.Server)
```
Comment (Umang01-hash) [inline pkg/gofr/datasource/pubsub/nats/client.go:36]: This log is a repeated in the Connect method called here:

```go
connManager := NewConnectionManager(c.Config, c.logger, c.natsConnector, c.jetStreamCreator)
	if err := connManager.Connect(); err != nil {
		c.logger.Errorf("failed to connect to NATS server at %v: %v", c.Config.Server, err)
		return err
	}
```


And inside the Connect method we also have this log:



```go
// Connect establishes a connection to NATS and sets up JetStream.
func (cm *ConnectionManager) Connect() error {
	cm.logger.Debugf("Connecting to NATS server at %v", cm.config.Server)

	opts := []nats.Option{nats.Name("GoFr NATS JetStreamClient")}

	if cm.config.CredsFile != "" {
		opts = append(opts, nats.UserCredentials(cm.config.CredsFile))
	}

	connInterface, err := cm.natsConnector.Connect(cm.config.Server, opts...)
	if err != nil {
		cm.logger.Errorf("failed to connect to NATS server at %v: %v", cm.config.Server, err)

		return err
	}
```

We need to remove either of them

**Review by Umang01-hash — CHANGES_REQUESTED** [review]
I think to maintain code uniformity isn't it better to introduce `Debugf` in interface wherever needed instead of using **Debug(fmt.Sprintf())** ?

**Review by Umang01-hash — APPROVED** [review]

**Review by vipul-rawat — APPROVED** [review]

**Comment by vikash** [comment]
If we have introduced debugf, we should check if we still need debug there. If not, we should probably remove it. @coolwednesday

**Review by Umang01-hash — APPROVED** [review]

**Review by vipul-rawat — APPROVED** [review]

**Pull request merged**